docs/platform/web-project.md

```diff
@@ -30,72 +30,143 @@ a _feature_ grouping it together with its dependencies,
+[service](https://www.ibm.com/developerworks/library/os-osgiblueprint/#servman) that implements the resource 
+mapping from the [whiteboard extender](http://ops4j.github.io/pax/web/SNAPSHOT/User-Guide.html#WhiteboardExtender-Howdoesithelponresourceregistration). For example:
 
-The client-side dependencies of your package — declared in the `package.json` file — 
-must also be provided to the platform as bundles.
+```xml
+<service interface="org.ops4j.pax.web.extender.whiteboard.ResourceMapping">
+  <bean class="org.ops4j.pax.web.extender.whiteboard.runtime.DefaultResourceMapping">
+    <property name="alias" value="/urlWhereResourceWillBeAvailable"/>
+    <property name="path" value="pathRelativeToBundleRootThatContainsResources"/>
+  </bean>
+</service>
+```
 
-If they are third-party code that means you would have to bundle them yourself, creating separate modules, etc..
+Finally, the Javascript source code is to be placed in `impl/src/main/javascript/web`. The `web` 
+folder name is a convention but the project creator may change it as long as the value of 
+the `path` property for the resource mapping service is changed accordingly in the blueprint descriptor.
 
-Luckily there is a project that already packages client-side web libraries as JAR files. 
-It is appropriately called [WebJars](http://www.webjars.org).
 
-All you have to do is to look for the needed libraries (the [NPM flavor](http://www.webjars.org/npm) is recommended), 
-choose the right versions and copy their Maven artifact information (groupId, artifactId and version).
+### Provisioning modules
 
-If you can't find the library, or the right version, you can create a new WebJar (light blue button on the top right corner).
+Consider the following example folder structure for `assemblies`:
 
-With the artifact information you can add the dependency to your feature definition.
-Just build the Maven artifact URL in the form `mvn:GROUP_ID/ARTIFACT_ID/VERSION`.
+```
+|____rootProject
+| |____assemblies
+| | |____src
+| | | |____main
+| | | | |____feature
+```
 
-However, WebJars are just plain JAR files without the manifest headers needed to make it an OSGi bundle.
-The Pentaho platform provides an Apache Karaf deployer that solves the problem:
-just prepend `pentaho-webjars:` to the artifact URL.
+As it was the case for the `impl` folder example, here we have a single module for provisioning located 
+directly in the `assemblies` folder. It is this module that is responsible for building
+the KAR file for deploy.
 
-In the end the bundle description in the feature file will look like this:
+Not only your bundle but also the client-side dependencies of your package — declared in the 
+[`package.json`](./web-package) file — must be provided to the platform as bundles. 
+
+If they are third-party code that means you would have to bundle them yourself, creating separate modules, etc. 
+Luckily there is a project that already packages client-side web libraries as JAR files. 
```

Comment:
Added the comma and used @carlosrusso suggestion of "Fortunately" instead of "Luckily"